pkg/git/factory/gitfactory.go

```diff
@@ -2,43 +2,81 @@ package gitfactory
 
 import (
 	"context"
+	"fmt"
+	"path/filepath"
 
+	"github.com/aws/eks-anywhere/pkg/addonmanager/addonclients"
 	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
 	"github.com/aws/eks-anywhere/pkg/git"
+	"github.com/aws/eks-anywhere/pkg/git/gitclient"
 	"github.com/aws/eks-anywhere/pkg/git/gogithub"
 	"github.com/aws/eks-anywhere/pkg/git/providers/github"
 )
 
-type gitProviderFactory struct {
-	GitClient github.GitClient
-}
+func Build(ctx context.Context, cluster *v1alpha1.Cluster, fluxConfig *v1alpha1.FluxConfig, writer filewriter.FileWriter) (*addonclients.GitTools, error) {
+	var clientOptions []gitclient.Opt
+	var repo string
+	var err error
 
-type Options struct {
-	GithubGitClient github.GitClient
-}
+	tools := &addonclients.GitTools{}
 
-func New(opts Options) *gitProviderFactory {
-	return &gitProviderFactory{GitClient: opts.GithubGitClient}
-}
+	switch {
+	case fluxConfig.Spec.Github != nil:
+		githubToken, err := github.GetGithubAccessTokenFromEnv()
+		if err != nil {
+			return tools, err
+		}
+
+		repo = fluxConfig.Spec.Github.Repository
+		clientOptions = append(clientOptions, gitclient.WithRepositoryUrl(github.RepoUrl(fluxConfig.Spec.Github.Owner, repo)))
+
+		clientAuth := git.TokenAuth{Token: githubToken, Username: fluxConfig.Spec.Github.Owner}
+		clientOptions = append(clientOptions, gitclient.WithTokenAuth(clientAuth))
+
+		tools.Provider, err = buildGithubProvider(ctx, clientAuth, fluxConfig.Spec.Github.Owner, repo, fluxConfig.Spec.Github.Personal)
+		if err != nil {
+			return tools, fmt.Errorf("building github provider: %v", err)
+		}
+	default:
+		return tools, fmt.Errorf("no valid git provider in FluxConfigSpec. Spec: %v", fluxConfig)
+	}
+
+	localGitRepoPath := filepath.Join(cluster.Name, "git", repo)
+	clientOptions = append(clientOptions, gitclient.WithRepositoryDirectory(localGitRepoPath))
+	tools.Client = gitclient.New(clientOptions...)
```

Comment:
should we maybe have a separate method that returns `client` that completely isolates git client config from provider?
```
func gitClient(...) Client {
  clientOptions := []gitclient.Opt{
    gitclient.WithRepositoryUrl(),
    gitclient.WithTokenAuth(),
    gitclient.WithRepositoryDirectory()),
  }
  
  return gitclient.New(clientOptions...)
}
```

Reply:
we can move it to a separate method, but we can't completely isolate it -- there are provider-specific configurations that the client will receive, such as the repo URL and the kind of auth

Reply:
we can do something like 1e37fe5fc308bbcdaa8fa675a830cb9bed482244

Reply:
the issue is that the auth is going to be provider dependent; the repo URL is provider dependent; and we can either have the git client options configured here, as we processes the provider, or we can have it itself review the FluxConfig again and construct the values; either way, we need to be able to have some level of coordination between what the provider is and what the gitClient is

Reply:
ok further isolated this --9efcc6c0

Reply:
that method will change a bit once we introduce ssh auth in the next pr